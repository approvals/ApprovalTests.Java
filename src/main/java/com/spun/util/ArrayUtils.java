package com.spun.util;

import java.lang.reflect.Array;
import java.lang.reflect.Method;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Iterator;
import java.util.List;
import java.util.Vector;

import org.lambda.functions.Function1;

/**
 * A static class of convenience methods for arrays and collections.
 **/
public class ArrayUtils
{
  /************************************************************************/
  public static <T> java.util.Collection<T> addArray(java.util.Collection<T> v, T array[])
  {
    if ((array == null) || (v == null)) { return v; }
    for (int i = 0; i < array.length; i++)
    {
      v.add(array[i]);
    }
    return v;
  }
  public static <T> String toString(T[] values, Function1<T, String> formatter)
  {
    return toString(Arrays.asList(values), formatter);
  }
  public static <T> String toString(Iterable<T> values, Function1<T, String> formatter)
  {
    StringBuffer b = new StringBuffer();
    for (T t : values)
    {
      b.append(formatter.call(t) + "\r\n");
    }
    return b.toString();
  }
  /***********************************************************************/
  public static Vector toReverseVector(Vector<Object> vector)
  {
    Vector<Object> reverse = new Vector<Object>(vector.size());
    for (int i = vector.size() - 1; i >= 0; i--)
    {
      reverse.add(vector.elementAt(i));
    }
    return reverse;
  }
  /***********************************************************************/
  public static <T> T[] toReverseArray(T[] array)
  {
    for (int i = 0; i < array.length / 2; i++)
    {
      T o1 = array[i];
      int end = array.length - i - 1;
      T o2 = array[end];
      array[i] = o2;
      array[end] = o1;
    }
    return array;
  }
  /***********************************************************************/
  public static <T> T[] addToArray(T[] array, T object)
  {
    Object[] newArray = null;
    if (array == null)
    {
      newArray = (Object[]) Array.newInstance(object.getClass(), 1);
      newArray[0] = object;
    }
    else
    {
      newArray = (Object[]) Array.newInstance(array.getClass().getComponentType(), array.length + 1);
      System.arraycopy(array, 0, newArray, 0, array.length);
      newArray[array.length] = object;
    }
    return (T[]) newArray;
  }
  /***********************************************************************/
  public static <T> T[] getSubsection(T[] array, int startInclusive, int endExclusive)
  {
    int length = endExclusive - startInclusive;
    length = length > array.length ? array.length : length;
    T[] newArray = (T[]) Array.newInstance(array.getClass().getComponentType(), length);
    System.arraycopy(array, startInclusive, newArray, 0, length);
    return newArray;
  }
  /***********************************************************************/
  public static boolean isEmpty(Object[] array)
  {
    return ((array == null) || (array.length == 0));
  }
  /************************************************************************/
  public static boolean isEmpty(Collection collection)
  {
    return ((collection == null) || (collection.size() == 0));
  }
  /***********************************************************************/
  public static <T> T getSingleton(T[] parts)
  {
    if (parts == null) { return null; }
    switch (parts.length)
    {
      case 0 :
        return null;
      case 1 :
        return parts[0];
      default :
        throw new Error("Called with more than one object in the array " + Arrays.asList(parts));
    }
  }
  /***********************************************************************/
  public static <H, T extends H> T getFirst(T[] array, Comparator<H> compartor)
  {
    return get(array, compartor, true);
  }
  /***********************************************************************/
  public static <H, T extends H> T getFirst(Collection<T> array, Comparator<H> sorter)
  {
    return get((T[]) array.toArray(), sorter, true);
  }
  /***********************************************************************/
  public static <T> T getLast(T[] array, Comparator<T> sorter)
  {
    return get(array, sorter, false);
  }
  /***********************************************************************/
  public static <T> T getLast(Collection<T> array, Comparator<T> sorter)
  {
    return get((T[]) array.toArray(), sorter, false);
  }
  /***********************************************************************/
  private static <H, T extends H> T get(T[] array, Comparator<H> sorter, boolean wantFirst)
  {
    if (isEmpty(array)) { return null; }
    T last = array[0];
    for (int i = 1; i < array.length; i++)
    {
      int compare = sorter.compare(last, array[i]);
      if ((wantFirst && compare > 0) || (!wantFirst && compare < 0))
      {
        last = array[i];
      }
    }
    return last;
  }
  /***********************************************************************/
  public static List combineResults(Object[] array, String invokeMethod)
  {
    if (ArrayUtils.isEmpty(array)) { return Collections.EMPTY_LIST; }
    try
    {
      return combineResults(array, ObjectUtils.getGreatestCommonDenominator(array, invokeMethod));
    }
    catch (Exception e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
  /***********************************************************************/
  @SuppressWarnings("rawtypes")
  public static List combineResults(Object[] array, Method method)
  {
    if (isEmpty(array)) { return new ArrayList(0); }
    try
    {
      ArrayList<Object> list = new ArrayList<Object>();
      for (int i = 0; i < array.length; i++)
      {
        addArray(list, (Object[]) method.invoke(array[i], (Object[]) null));
      }
      return list;
    }
    catch (Throwable e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
  /***********************************************************************/
  public static Object[] combine(Object[] a, Object[] b)
  {
    if (isEmpty(a) && isEmpty(b))
    {
      if (a != null)
      {
        return (Object[]) Array.newInstance(a.getClass().getComponentType(), 0);
      }
      else if (b != null)
      {
        return (Object[]) Array.newInstance(b.getClass().getComponentType(), 0);
      }
      else
      {
        return new Object[0];
      }
    }
    else if (isEmpty(a))
    {
      return b;
    }
    else if (isEmpty(b))
    {
      return a;
    }
    else
    {
      Object[] newArray = null;
      newArray = (Object[]) Array.newInstance(a.getClass().getComponentType(), a.length + b.length);
      System.arraycopy(a, 0, newArray, 0, a.length);
      System.arraycopy(b, 0, newArray, a.length, b.length);
      return newArray;
    }
  }
  /***********************************************************************/
  public static <T> boolean contains(T[] values, T value)
  {
    for (int i = 0; i < values.length; i++)
    {
      if (value.equals(values[i])) { return true; }
    }
    return false;
  }
  /***********************************************************************/
  public static boolean contains(int[] values, int value)
  {
    for (int i = 0; i < values.length; i++)
    {
      if (value == values[i]) { return true; }
    }
    return false;
  }
  /************************************************************************/
  public static <T> T getLast(List<T> list)
  {
    return list.get(list.size() - 1);
  }
  /************************************************************************/
  public static <K, T> T getDefault(HashMap<K, T> map, K key, T defaultValue)
  {
    T value = map.get(key);
    if (value == null)
    {
      map.put(key, defaultValue);
      value = defaultValue;
    }
    return value;
  }
  /************************************************************************/
  public static <K, V> int countValues(HashMap<K, V> out, V matching)
  {
    return count(matching, out.values());
  }
  /************************************************************************/
  public static <V> int count(V matching, Collection<V> values)
  {
    int count = 0;
    for (V value : values)
    {
      if (matching.equals(value))
      {
        count++;
      }
    }
    return count;
  }
  public static <T> List<T> combine(List<T> list1, List<T> list2)
  {
    List<T> all = new ArrayList<T>();
    all.addAll(list1);
    all.addAll(list2);
    return all;
  }
  public static <T> Iterable<T> asIterable(Iterator<T> iterator)
  {
    return new IterableWrapper(iterator);
  }
  public static class IterableWrapper<T> implements Iterable<T>
  {
    private final Iterator<T> iterator;
    public IterableWrapper(Iterator<T> iterator)
    {
      this.iterator = iterator;
    }
    @Override
    public Iterator<T> iterator()
    {
      return iterator;
    }
  }
}