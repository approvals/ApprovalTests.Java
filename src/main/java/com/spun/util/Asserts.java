package com.spun.util;

import java.io.File;

public class Asserts
{
  public static void t()
  {
    // can you see it now?  good.
    // how about this?  
    // kind of slow...
  }
  /***********************************************************************/
  public static <T> T assertNotNull(String label, T o)
  {
    if (o == null) { throw new NullPointerException(label + " was null"); }
    return o;
  }
  /***********************************************************************/
  public static void assertEqual(String label, int expected, int actual)
  {
    if (expected != actual) { throw new IllegalStateException(label + " " + expected + "!=" + actual); }
  }
  /***********************************************************************/
  public static void assertFileDoesNotExist(String label, String fileName)
  {
    File file = new File(fileName);
    if (file.exists()) { throw new IllegalStateException(label + " file " + file.getAbsolutePath()
        + " already exists"); }
  }
  /***********************************************************************/
  public static void assertFileExists(String label, String fileName)
  {
    File file = new File(fileName);
    if (!file.exists()) { throw new IllegalStateException(label + " file " + file.getAbsolutePath()
        + " does not exist"); }
  }
  /***********************************************************************/
  public static void equals(String label, double expected, double actual, double percission)
  {
    if (!NumberUtils.equals(expected, actual, percission)) { throw new IllegalStateException(label + " "
        + expected + "!=" + actual); }
  }
  /***********************************************************************/
  public static void notEquals(String label, double expected, double actual)
  {
    if (actual == expected) { throw new IllegalStateException(label + " " + expected + "!=" + actual); }
  }
  /***********************************************************************/
  public static void assertEqual(String label, Object expected, Object actual)
  {
    if (!ObjectUtils.isEqual(expected, actual)) { throw new IllegalStateException(label + " " + expected + "!="
        + actual); }
  }
  /***********************************************************************/
  public static void assertNotEqual(String label, Object expected, Object actual)
  {
    if (ObjectUtils.isEqual(expected, actual)) { throw new IllegalStateException(label + " can't = " + expected); }
  }
  /***********************************************************************/
  public static void assertEqual(String label, double expected, double actual, double precision)
  {
    if (!NumberUtils.equals(expected, actual, precision)) { throw new IllegalStateException(label + " " + expected
        + "!=" + actual + "within delta of" + precision); }
  }
  /***********************************************************************/
  public static void assertStartsWith(String label, String haystack, String needle)
  {
    if (haystack == null || !haystack.startsWith(needle)) { throw new IllegalStateException(label + " " + haystack
        + " does not start with " + needle); }
  }
  /***********************************************************************/
  /***********************************************************************/
  public static void assertTrue(String label, boolean b)
  {
    if (!b) { throw new IllegalStateException(label + " was false"); }
  }
}
