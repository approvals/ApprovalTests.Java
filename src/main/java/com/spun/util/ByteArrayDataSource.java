package com.spun.util;

/*
 * @(#)ByteArrayDataSource.java	1.1 00/01/30
 *
 * Copyright 1998-2000 Sun Microsystems, Inc. All Rights Reserved.
 * 
 * This software is the proprietary information of Sun Microsystems, Inc.  
 * Use is subject to license terms.
 * 
 */
import java.io.*;
import javax.activation.*;

/**
 * Comment By LLewellyn: This is a needed class for using mail. For some reason wasn't 
 * made as part of the standard mail package.<P>
 *
 * A simple DataSource for demonstration purposes.
 * This class implements a DataSource from:
 *   an InputStream
 *  a byte array
 *   a String
 *
 * @author John Mani
 * @author Bill Shannon
 * @author Max Spivak
 **/
public class ByteArrayDataSource implements DataSource
{
  private byte[] data;          // data
  private String type;          // content-type
  private String name = "dummy";
  /***********************************************************************/
  /* Create a DataSource from an input stream */
  public ByteArrayDataSource(InputStream is, String type)
  {
    this(is, type, null);
  }
  /***********************************************************************/
  /* Create a DataSource from an input stream */
  public ByteArrayDataSource(InputStream is, String type, String name)
  {
    this.type = type;
    setName(name);
    try
    {
      ByteArrayOutputStream os = new ByteArrayOutputStream();
      int ch;
      while ((ch = is.read()) != -1)
      {
        // XXX - must be made more efficient by
        // doing buffered reads, rather than one byte reads
        os.write(ch);
      }
      data = os.toByteArray();
    }
    catch (IOException ioex)
    {
    }
  }
  /***********************************************************************/
  /* Create a DataSource from a byte array */
  public ByteArrayDataSource(byte[] data, String type)
  {
    this.data = data;
    this.type = type;
  }
  /***********************************************************************/
  public void setName(String name)
  {
    this.name = (name == null) ? "dummy" : name;
  }
  /***********************************************************************/
  /* Create a DataSource from a String */
  public ByteArrayDataSource(String data, String type)
  {
    this(data, type, null);
  }
  /***********************************************************************/
  /* Create a DataSource from a String */
  public ByteArrayDataSource(String data, String type, String name)
  {
    setName(name);
    try
    {
      // Assumption that the string contains only ASCII
      // characters!  Otherwise just pass a charset into this
      // constructor and use it in getBytes()
      this.data = data.getBytes("iso-8859-1");
    }
    catch (UnsupportedEncodingException uex)
    {
    }
    this.type = type;
  }
  /***********************************************************************/
  /**
   * Return an InputStream for the data.
   * Note - a new stream must be returned each time.
   */
  public InputStream getInputStream() throws IOException
  {
    if (data == null) { throw new IOException("no data"); }
    return new ByteArrayInputStream(data);
  }
  /***********************************************************************/
  public OutputStream getOutputStream() throws IOException
  {
    throw new IOException("cannot do this");
  }
  /***********************************************************************/
  public String getContentType()
  {
    return type;
  }
  /***********************************************************************/
  public String getName()
  {
    return name;
  }
  /***********************************************************************/
  /***********************************************************************/
}