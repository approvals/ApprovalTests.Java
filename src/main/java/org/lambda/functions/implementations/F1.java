package org.lambda.functions.implementations;

public class F1<In, Out> extends Function<In, Object, Object, Object, Object, Object, Object, Object, Object, Out>
    implements
      org.lambda.functions.Function1<In, Out>
{
  public F1(In a, Object... extraVariables)
  {
    super(extraVariables);
    this.a = a;
  }
  public Out call(In i)
  {
    return call(new Object[]{i});
  }
}
