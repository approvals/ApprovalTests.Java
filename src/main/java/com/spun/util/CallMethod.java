package com.spun.util;

import java.util.ArrayList;

public class CallMethod
{
  private final Class<? extends Object>             clazz;
  private ArrayList<String>                         methodNames = new ArrayList<String>();
  private ArrayList<MethodExecutionPath.Parameters> params      = new ArrayList<MethodExecutionPath.Parameters>();
  public CallMethod(Class<? extends Object> clazz)
  {
    this.clazz = clazz;
  }
  public CallMethod method(String methodName, Object... parameters)
  {
    methodNames.add(methodName);
    if (!ArrayUtils.isEmpty(parameters))
    {
      params.add(new MethodExecutionPath.Parameters(parameters));
    }
    else {
      params.add(new MethodExecutionPath.Parameters((Object[])null));
    }
    return this;
  }
  public MethodExecutionPath go()
  {
    return new MethodExecutionPath(clazz, methodNames.toArray(new String[0]), params.toArray(new MethodExecutionPath.Parameters[0]));
  }
}
