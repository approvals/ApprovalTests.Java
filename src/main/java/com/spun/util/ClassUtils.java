package com.spun.util;

import java.io.File;
import java.lang.reflect.Field;
import java.lang.reflect.Modifier;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ClassUtils
{
  /************************************************************************/
  public static String getClassName(Class<?> clazz)
  {
    String name = clazz.getName();
    int divider = name.lastIndexOf(".");
    return name.substring(divider + 1);
  }
  /************************************************************************/
  public static String getClassPath(Class<?> clazz)
  {
    String name = clazz.getName();
    int divider = name.lastIndexOf(".");
    return name.substring(0, divider);
  }
  /************************************************************************/
  public static Class<?> getWrapperClass(Class<?> primativeType)
  {
    if (boolean.class.equals(primativeType))
    {
      return Boolean.class;
    }
    else if (float.class.equals(primativeType))
    {
      return Float.class;
    }
    else if (long.class.equals(primativeType))
    {
      return Long.class;
    }
    else if (int.class.equals(primativeType))
    {
      return Integer.class;
    }
    else if (short.class.equals(primativeType))
    {
      return Short.class;
    }
    else if (byte.class.equals(primativeType))
    {
      return Byte.class;
    }
    else if (double.class.equals(primativeType))
    {
      return Double.class;
    }
    else if (char.class.equals(primativeType))
    {
      return Character.class;
    }
    else
    {
      return primativeType;
    }
  }
  /************************************************************************/
  /************************************************************************/
  public static boolean hasMethod(Class<?> clazz, String methodName, Class<?>... parameterTypes)
  {
    try
    {
      return clazz.getMethod(methodName, parameterTypes) != null;
    }
    catch (SecurityException e)
    {
      throw ObjectUtils.throwAsError(e);
    }
    catch (NoSuchMethodException e)
    {
      return false;
    }
  }
  public static File getSourceDirectory(Class<?> clazz)
  {
    final String name = clazz.getName();
    String[] split = name.split("\\.");
    split[split.length - 1] = split[split.length - 1] + ".java";
    File found = find(new File("."), Arrays.asList(split));
    if (found == null) { throw new FormattedException("Didn't find %s under %s", name, new File(".")); }
    return found.getParentFile();
  }
  public static File find(File file, List<String> matches)
  {
    ArrayList<String> copy = new ArrayList<String>();
    copy.addAll(matches);
    copy.add(0, "*");
    return find2(file, copy);
  }
  public static File find2(File file, List<String> matches)
  {
    SearchingFileFilter searchingFileFilter = new SearchingFileFilter(matches);
    File[] listFiles = file.listFiles(searchingFileFilter);
    for (File file2 : listFiles)
    {
      if (file2.isDirectory())
      {
        File found = find2(file2, searchingFileFilter.getSubset(file2.getName()));
        if (found != null) { return found; }
      }
      else
      {
        return file2;
      }
    }
    return null;
  }
  public static <T> T create(Class<T> clazz)
  {
    try
    {
      return clazz.newInstance();
    }
    catch (Exception e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
  public static boolean IsPublicStatic(Field field)
  {
      return field != null && Modifier.isPublic(field.getModifiers()) && Modifier.isStatic(field.getModifiers());
  }
  public static boolean isPrimitiveField(Field field)
  {
    return field != null && (field.getType().isArray() || field.getType().isPrimitive() || field.getType().isAssignableFrom(String.class));
  }
}
