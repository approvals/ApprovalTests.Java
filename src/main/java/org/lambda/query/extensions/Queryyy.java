package org.lambda.query.extensions;

import java.util.List;

import org.lambda.functions.Function1;
import org.lambda.query.Query;

public class Queryyy<In> extends ExtendableBase<List<In>>
{
  public <Out> List<Out> select(Function1<In, Out> function)
  {
    return Query.select(caller, function);
  }
  public List<In> where(Function1<In, Boolean> function)
  {
    return Query.where(caller, function);
  }
  public In max(Function1<In, Comparable> function)
  {
    return Query.max(caller, function);
  }
  public In min(Function1<In, Comparable> function)
  {
    return Query.min(caller, function);
  }
}
