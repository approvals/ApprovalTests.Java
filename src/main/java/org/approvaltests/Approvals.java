package org.approvaltests;

import java.awt.Component;
import java.awt.Image;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.FileFilter;
import java.io.FilenameFilter;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

import org.apache.hadoop.fs.Path;
import org.approvaltests.ReporterFactory.FileTypes;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.core.ApprovalFailureOverrider;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.ApprovalWriter;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.JUnitStackTraceNamer;
import org.approvaltests.reporters.ExecutableQueryFailure;
import org.approvaltests.writers.ApprovalBinaryFileWriter;
import org.approvaltests.writers.ApprovalTextWriter;
import org.approvaltests.writers.ApprovalXmlWriter;
import org.approvaltests.writers.ComponentApprovalWriter;
import org.approvaltests.writers.DirectoryToDirectoryWriter;
import org.approvaltests.writers.FileApprovalWriter;
import org.approvaltests.writers.ImageApprovalWriter;
import org.approvaltests.writers.ResultSetApprovalWriter;
import org.jrack.RackResponse;
import org.jrack.RackResponseUtils;
import org.lambda.functions.Function1;
import org.lambda.functions.implementations.F1;
import org.lambda.query.Query;

import com.spun.util.ArrayUtils;
import com.spun.util.ObjectUtils;
import com.spun.util.StringUtils;
import com.spun.util.images.ImageWriter;
import com.spun.util.persistence.ExecutableQuery;
import com.spun.util.persistence.SqlLoader;

public class Approvals
{
  /**
   * @deprecated Use verify()
   */
  public static void approve(String response) throws Exception
  {
    verify(response);
  }
  public static void verify(String response) throws Exception
  {
    verify(new ApprovalTextWriter(response, "txt"), FileTypes.Text);
  }
  public static void verify(Object o) throws Exception
  {
    verify("" + o);
  }
  /**
   * @deprecated Use verifyAll()
   */
  public static <T> void approve(String label, T[] array) throws Exception
  {
    verifyAll(label, array);
  }
  public static <T> void verifyAll(String label, T[] array) throws Exception
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"), FileTypes.Text);
  }
  /**
   * @deprecated Use verifyAll()
   */
  public static <T> void approve(String header, String label, T[] array) throws Exception
  {
    verifyAll(header, label, array);
  }
  public static <T> void verifyAll(String header, String label, T[] array) throws Exception
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"),
        FileTypes.Text);
  }
  /**
   * @deprecated Use verifyAll()
   */
  public static <T> void approve(T[] values, Function1<T, String> f1)
  {
    verifyAll(values, f1);
  }
  public static <T> void verifyAll(T[] values, Function1<T, String> f1)
  {
    String text = ArrayUtils.toString(values, f1);
    verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
  }
  /**
   * @deprecated Use verifyAll()
   */
  public static <T> void approve(String header, T[] values, Function1<T, String> f1)
  {
    verifyAll(header, values, f1);
  }
  public static <T> void verifyAll(String header, T[] values, Function1<T, String> f1)
  {
    verifyAll(header, Arrays.asList(values), f1);
  }
  public static <T> void verifyAll(String header, Iterable<T> array, Function1<T, String> f1)
  {
    String text = formatHeader(header) + ArrayUtils.toString(array, f1);
    verify(new ApprovalTextWriter(text, "txt"), FileTypes.Text);
  }
  private static String formatHeader(String header)
  {
    return StringUtils.isEmpty(header) ? "" : header + "\r\n\r\n\r\n";
  }
  /**
   * @deprecated Use verifyAll()
   */
  public static <T> void approve(String label, Iterable<T> array) throws Exception
  {
    verifyAll(label, array);
  }
  public static <T> void verifyAll(String label, Iterable<T> array) throws Exception
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"), FileTypes.Text);
  }
  /**
   * @deprecated Use verifyAll()
   */
  public static <T> void approve(String header, String label, Iterable<T> array) throws Exception
  {
    verifyAll(header, label, array);
  }
  public static <T> void verifyAll(String header, String label, Iterable<T> array) throws Exception
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"),
        FileTypes.Text);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(Component c) throws Exception
  {
    verify(c);
  }
  public static void verify(Component c) throws Exception
  {
    verify(new ComponentApprovalWriter(c), FileTypes.Image);
  }
  /**
   * @deprecated Use verifyHtml()
   */
  public static void approveHtml(String response) throws Exception
  {
    verifyHtml(response);
  }
  public static void verifyHtml(String response) throws Exception
  {
    verify(new ApprovalTextWriter(response, "html"), FileTypes.Html);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(File generateFile)
  {
    verify(generateFile);
  }
  public static void verify(File generateFile)
  {
    verify(new FileApprovalWriter(generateFile), FileTypes.File);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(Image image)
  {
    verify(image);
  }
  public static void verify(Image image)
  {
    approve(ImageWriter.toBufferedImage(image), createApprovalNamer());
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(BufferedImage bufferedImage)
  {
    verify(bufferedImage);
  }
  public static void verify(BufferedImage bufferedImage)
  {
    verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(ApprovalWriter writter, ApprovalNamer namer, ApprovalFailureReporter reporter)
  {
    verify(writter, namer, reporter);
  }
  public static void verify(ApprovalWriter writter, ApprovalNamer namer, ApprovalFailureReporter reporter)
  {
    verify(new FileApprover(writter, namer), reporter);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(ApprovalWriter writter, String fileType)
  {
    verify(writter, fileType);
  }
  public static void verify(ApprovalWriter writter, String fileType)
  {
    verify(writter, createApprovalNamer(), ReporterFactory.get(fileType));
  }
  /**
   * @deprecated Use verifyXml()
   */
  public static void approveXml(String xml) throws Exception
  {
    verifyXml(xml);
  }
  public static void verifyXml(String xml) throws Exception
  {
    verify(new ApprovalXmlWriter(xml), FileTypes.Text);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(FileApprover approver, ApprovalFailureReporter reporter)
  {
    verify(approver, reporter);
  }
  public static void verify(FileApprover approver, ApprovalFailureReporter reporter)
  {
    try
    {
      if (!approver.approve())
      {
        boolean passed = false;
        if (reporter instanceof ApprovalFailureOverrider)
        {
          passed = approver.askToChangeReceivedToApproved((ApprovalFailureOverrider) reporter);
        }
        if (!passed)
        {
          approver.reportFailure(reporter);
          approver.fail();
        }
        else
        {
          approver.cleanUpAfterSuccess(reporter);
        }
      }
      else
      {
        approver.cleanUpAfterSuccess(reporter);
      }
    }
    catch (Exception e)
    {
      throw ObjectUtils.throwAsError(e);
    }
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(ExecutableQuery query) throws Exception
  {
    verify(query);
  }
  public static void verify(ExecutableQuery query) throws Exception
  {
    verify(new ApprovalTextWriter(query.getQuery(), "txt"), new JUnitStackTraceNamer(),
        new ExecutableQueryFailure(query));
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(Map map) throws Exception
  {
    verify(map);
  }
  public static void verify(Map map) throws Exception
  {
    verify(new ApprovalTextWriter(StringUtils.toString(map), "txt"), FileTypes.Text);
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(RackResponse response) throws Exception
  {
    verify(response);
  }
  public static void verify(RackResponse response) throws Exception
  {
    if (isImage(response))
    {
      String fileType = "png";
      verify(new ApprovalBinaryFileWriter(response.getResponse(), fileType), fileType);
    }
    else
    {
      verifyHtml(response.getResponse().toString());
    }
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(ResultSet rs) throws Exception
  {
    verify(rs);
  }
  public static void verify(ResultSet rs) throws Exception
  {
    verify(new ResultSetApprovalWriter(rs), "csv");
  }
  /**
   * @deprecated Use verify()
   */
  public static void approve(SqlLoader loader) throws Exception
  {
    verify(loader);
  }
  public static void verify(SqlLoader loader) throws Exception
  {
    verify(new SqlLoader.ExecutableWrapper(loader));
  }
  private static boolean isImage(RackResponse response)
  {
    String type = response.getHeaders().get(RackResponseUtils.CONTENT_TYPE);
    return RackResponseUtils.CONTENT_TYPE_IMAGE.equals(type);
  }
  public static ApprovalNamer createApprovalNamer()
  {
    return new JUnitStackTraceNamer();
  }
  private static void approve(BufferedImage bufferedImage, ApprovalNamer namer)
  {
    verify(new ImageApprovalWriter(bufferedImage), FileTypes.Image);
  }
  public static void verifyEachFileInDirectory(File directory)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles());
  }
  public static void verifyEachFileInDirectory(File directory, FileFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  public static void verifyEachFileInDirectory(File directory, FilenameFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  private static void verifyEachFileAgainstMasterDirectory(File[] files) throws Error
  {
    ApprovalNamer namer = createApprovalNamer();
    String dirName = namer.getSourceFilePath() + Path.SEPARATOR + namer.getApprovalName() + ".Files";
    File approvedDirectory = new File(dirName);
    List<File> mismatched = new ArrayList<File>();
    for (File f : files)
    {
      if (!f.isDirectory())
      {
        try
        {
          verify(new DirectoryToDirectoryWriter(f, approvedDirectory), FileTypes.File);
        }
        catch (Throwable e)
        {
          mismatched.add(f);
        }
      }
    }
    if (!mismatched.isEmpty())
    {
      String message = "The Following Files did not match up: " + getFileNameList(mismatched);
      throw new Error(message);
    }
  }
  private static String getFileNameList(List<File> mismatched)
  {
    return Query.select(mismatched, new F1<File, String>(mismatched.get(1))
    {
      {
        ret(a.getName());
      }
    }).toString();
  }
}