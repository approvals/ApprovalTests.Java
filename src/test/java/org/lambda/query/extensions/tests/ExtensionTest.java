package org.lambda.query.extensions.tests;

import java.util.Arrays;
import java.util.List;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.extensionmethods.ExtendableUtilities;
import org.lambda.functions.implementations.F1;
import org.lambda.query.extensions.Queryyy;

public class ExtensionTest extends TestCase
{
  @SuppressWarnings("unchecked")
  public void testHighestDivisorOf3() throws Exception
  {
    if (ExtendableUtilities.isNotConfigured()) { return; }
    List<Integer> list = Arrays.asList(30, 46, 60);
    Approvals.verifyAll("", list.use(Queryyy.class).where(new F1<Integer, Boolean>(0)
    {
      {
        returnValue(a % 3 == 0);
      }
    }).use(Queryyy.class).select(new F1<Integer, String>(0)
    {
      {
        returnValue(Integer.toHexString(a));
      }
    }));
  }
}
