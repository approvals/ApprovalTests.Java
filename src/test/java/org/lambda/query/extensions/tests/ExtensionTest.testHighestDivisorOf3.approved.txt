[0] = 1e
[1] = 3c
